Sources/_NIOConcurrency/AsyncAwaitSupport.swift

```diff
@@ -0,0 +1,189 @@
+    public func connect(to address: SocketAddress, file: StaticString = #file, line: UInt = #line) async throws {
+        try await self.connect(to: address, file: file, line: line).get()
+    }
+
+    /// Shortcut for calling `write` and `flush`.
+    ///
+    /// - parameters:
+    ///     - data: the data to write
+    /// - returns: the future which will be notified once the `write` operation completes.
+    public func writeAndFlush(_ data: NIOAny, file: StaticString = #file, line: UInt = #line) async throws {
+        try await self.writeAndFlush(data, file: file, line: line).get()
+    }
+
+    /// Close the `Channel` and so the connection if one exists.
+    ///
+    /// - parameters:
+    ///     - mode: the `CloseMode` that is used
+    /// - returns: the future which will be notified once the operation completes.
+    public func close(mode: CloseMode = .all, file: StaticString = #file, line: UInt = #line) async throws {
+        try await self.close(mode: mode, file: file, line: line).get()
+    }
+
+    /// Trigger a custom user outbound event which will flow through the `ChannelPipeline`.
+    ///
+    /// - parameters:
+    ///     - event: the event itself.
+    /// - returns: the future which will be notified once the operation completes.
+    public func triggerUserOutboundEvent(_ event: Any, file: StaticString = #file, line: UInt = #line) async throws {
+        try await self.triggerUserOutboundEvent(event, file: file, line: line).get()
+    }
+}
+
+extension ChannelPipeline {
+    public func addHandler(_ handler: ChannelHandler,
+                           name: String? = nil,
+                           position: ChannelPipeline.Position = .last) async throws {
+        try await self.addHandler(handler, name: name, position: position).get()
+    }
+
+    public func removeHandler(_ handler: RemovableChannelHandler) async throws {
+        try await self.removeHandler(handler).get()
+    }
+
+    public func removeHandler(name: String) async throws {
+        try await self.removeHandler(name: name).get()
+    }
+
+    public func removeHandler(context: ChannelHandlerContext) async throws {
+        try await self.removeHandler(context: context).get()
+    }
+
+    public func context(handler: ChannelHandler) async throws -> ChannelHandlerContext {
+        return try await self.context(handler: handler).get()
+    }
+
+    public func context(name: String) async throws -> ChannelHandlerContext {
+        return try await self.context(name: name).get()
+    }
+
+    public func context<Handler: ChannelHandler>(handlerType: Handler.Type) async throws -> ChannelHandlerContext {
+        return try await self.context(handlerType: handlerType).get()
+    }
+
+    public func addHandlers(_ handlers: [ChannelHandler],
+                            position: ChannelPipeline.Position = .last) async throws {
+        try await self.addHandlers(handlers, position: position)
```

Comment:
```suggestion
        try await self.addHandlers(handlers, position: position).get()
```

Reply:
this should use the `async` version of `addHandlers`

Reply:
you're right, sorry! done